### DISCUSSION TITLE
Change how `UpdateXlfOnBuild` is done for PRs

### DISCUSSION THREAD

**Discussion by jsoref** [dotnet/fsharp#16216] [Ideas]
It'd be really nice (especially for people who can't manage to get dotnet 8 working on their arm macOS systems...) if for PRs, CI would in one place run with `UpdateXlfOnBuild=true`, perform a diff (showing the output in the log) and add a failure if there's a diff (so that it's still a failure condition).

The end result of this would be that if some hapless soul (like me) can't get `msbuild /t:UpdateXlf` to work, the CI would still fail, but it'd also provide a patch which could be applied to fix up the xlf files and thus fix up the PR so that they could continue.

**Comment by vzarytovskii** [comment]
You can run (in your PR) `/run xlf`, this should generate files and create a commit. I have never tried it though, should work (update from last main, I fixed a type there).

**Reply by jsoref** [reply]
Thanks. It'd be nice if the error message suggested that.
